oligo_designer_toolsuite/oligo_specificity_filter/_filter_exact_matches.py

```diff
@@ -101,35 +116,38 @@ def get_oligo_pair_hits(
         assert (
             sequence_type in options
         ), f"Sequence type not supported! '{sequence_type}' is not in {options}."
+        sequence_type_reverse_complement = options[0] if options[0] != sequence_type else options[1]
 
         # extract all the sequences
-        search_results = self._get_duplicated_sequences(sequence_type, oligo_database)
+        sequences = oligo_database.get_sequence_list(sequence_type=sequence_type)
+        sequences_rc = oligo_database.get_sequence_list(sequence_type=sequence_type_reverse_complement)
+        search_results = self._get_duplicated_sequences(sequences + sequences_rc)
```

Comment:
I think we should give the user a choice whether to filter duplicated sequences, sequences that are reverse complement to each other, or both (param in the `__init__`)

Reply:
The rationale behind this is that we have two application cases:
1. apply function -> using exact matches as filter: here we want to remove oligos which are duplicated in the database, i.e. we use the same forward sequences
2. oligo_pair_hits function -> using exact matches to get hits table for cross-hybridization filter. Here we have to make sure to find the exact matches among the reverse complements, hence, we match all forward sequences to all reverse complements